bin/install-theos

```diff
@@ -0,0 +1,598 @@
+	# Check for Xcode
+	XCODE="/Applications/Xcode.app/"
+	if ! [[ -d "$XCODE" && "$(ls -A "$XCODE")" ]]; then
+		error "Xcode is mandatory. The Command Line Tools package isn't sufficient for Theos to work. Xcode provides toolchains for all Apple platforms."
+		exit 3
+	fi
+
+	# Check for Homebrew
+	update "Preparing to install dependencies..."
+	if ! command -v brew &> /dev/null; then
+		read -p "Homebrew, which provides tools Theos depends on, is not installed. Would you like to have it installed for you? [y/n]" hbrew
+		if [[ "$hbrew" == "y" || "$hbrew" == "yes" || "$hbrew" == "Y" || "$hbrew" == "Yes" ]]; then
+			bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)" && update "Homebrew has been successfully installed!" || (error "Homebrew install command seems to have encountered an error. Please see the log above." && exit 5)
+		else
+			error "Homebrew provides tools Theos depends on and thus is mandatory. Please install Homebrew before proceeding with the installation."
+			exit 4
+		fi
+	fi
+
+	# Dependencies
+	brew install ldid xz && update "Dependencies have been successfully installed!" || (error "Dependency install command seems to have encountered an error. Please see the log above." && exit 6)
+
+	# Check for $THEOS env var
+	update "Checking for \$THEOS environment variable..."
+	if [[ "$THEOS" ]]; then
+		update "\$THEOS is already set to '$THEOS'. Nothing to do here."
+	else
+		update "\$THEOS has not been set. Setting now..."
+
+		if [[ "$SHELL_ENV" == "unknown" ]]; then
+			error "Current shell ($CSHELL) is unsupported by this installer. Please set the THEOS environment variable to '~/theos' manually before proceeding."
+			exit 7
+		fi
+
+		# Set $THEOS
+		echo "export THEOS=~/theos" >> "$SHELL_ENV"
+		source "$SHELL_ENV"
+		if [[ "$THEOS" ]]; then
+			update "\$THEOS has been successfully set to '$THEOS'!"
+		else
+			error "\$THEOS is still not set despite our efforts. Try restarting your shell and/or running the installer again. If that doesn't do the trick, you may have to set the THEOS environment variable to '~/theos' manually before proceeding."
+			exit 8
+		fi
+	fi
+
+	# Get Theos
+	update "Checking for Theos install..."
+	if [[ -d "$THEOS" && "$(ls -A "$THEOS")" ]]; then
+		update "Theos appears to already be installed. Checking for updates..."
+		$THEOS/bin/update-theos
+	else
+		update "Theos does not appear to be installed. Cloning now..."
+		git clone --recursive https://github.com/theos/theos.git $THEOS && update "Git clone of Theos was successful!" || (error "Theos git clone command seems to have encountered an error. Please see the log above." && exit 9)
+	fi
+
+	# Get patched sdks
+	update "Checking for patched sdks..."
+	if [[ -d "$THEOS/sdks/" && "$(ls -A "$THEOS/sdks/" | grep sdk)" ]]; then
+		update "SDKs appear to already be installed."
+	else
+		update "SDKS do not appear to be installed. Installing now..."
+		curl -LO https://github.com/theos/sdks/archive/master.zip
```

Comment:
Since we have quite a few places where we curl and unzip/untar something, could we centralise this to a function? May as well change to `master.tar.gz` so that function can just assume it needs to use tar.

Reply:
Is this still something you're interested in? 

The sdk curl has been moved to its own function to avoid duplication for each platform and the only other curls are the Homebrew install script (for those who don't have it installed and wish for it to be) and the non-Darwin toolchains (2xLinux and 2xFreeBSD) which can be its own function if you want to slim that down.